includes/promoted-jobs/class-wp-job-manager-promoted-jobs-api.php

```diff
@@ -62,6 +62,31 @@ public function register_routes() {
 				],
 			]
 		);
+		register_rest_route(
+			self::NAMESPACE,
+			self::REST_BASE . '/(?P<job_id>[\d]+)',
+			[
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_job_data' ],
+					'permission_callback' => '__return_true',
+					'args'                => [
+						'job_id'  => [
+							'required' => true,
+							'type'     => 'integer',
+						],
+						'user_id' => [
+							'required' => true,
+							'type'     => 'integer',
+						],
+						'token'   => [
```

Comment:
I'm thinking if it would make more sense to include the token in the header instead of in the body. But honestly, I don't see any benefit in changing that. I was just thinking that it could make more sense, so feel free to ignore it if you disagree.

Reply:
I think the main issue with that is that, by making it a header, we might cause issues with reverse proxy (nginx/varnish/cloudflare, maybe?), and WP wouldn't enforce it as well as it does right now.

If this was on WPJMCOM side, I think it would be ok, though. :thinking: